core/dbt/parser/schema_generic_tests.py

```diff
@@ -361,7 +363,11 @@ def parse_test(
         column: Optional[UnparsedColumn],
         version: Optional[NodeVersion],
     ) -> None:
+
         if isinstance(test, str):
+            # warn if test name has trailing characters which are unused
+            if len(test.split()) > 1:
+                warn_or_error(UnusedTrailingYaml(test=test))
```

Comment:
@dbeatty10 do you recommend putting the `warn_or_error` call somewhere else? I'd like to reference the node name in the warning, rather than the str `test`, but wasn't sure where to check for the issue after parsing.

Reply:
I don't have a good sense of this one way or the other.

But you could look through other locations where `warn_or_error` is utilized to see if any of those serve as good guidance.

Here's a listing:
https://github.com/search?q=repo%3Adbt-labs%2Fdbt-core%20warn_or_error&type=code

You'd probably want to start by looking at the examples within `core/dbt/parser/` and see if any of them are relevant to your use-case.

Reply:
I scoured the parser folder and don't see a better place for it. 

Really my comment was about if specifying the test name, and not the node name, would suffice. I think it would. Using the example from the issue the output would be:
```
[WARNING]: The test "not_null *where_filter" has trailing characters that will be ignored. If those characters are meaningful append a ":" after "unique". 
```